app/code/Magento/CatalogInventory/Observer/ReindexQuoteInventoryObserver.php

```diff
@@ -63,7 +77,11 @@ public function execute(EventObserver $observer)
         }
 
         if ($productIds) {
-            $this->stockIndexerProcessor->reindexList($productIds);
+            try {
+                $this->stockIndexerProcessor->reindexList($productIds);
+            } catch (\Exception $exception) {
+                $this->logger->error($exception);
```

Comment:
Actually you just hide the error, but the indexer still failing

Reply:
Why the elastic is triggered by stock indexer ?

Reply:
Hi @Nazar65, I believe the stock indexer triggers the `Magento\Catalog\Model\Indexer\Product\Price\Processor` and for that reason, it is relaying on ElasticSearch. We could look into some decomposition in this case, but I guess the price indexer would still be something that might be needed to run.

Magento depends on ElasticSearch to work properly, Catalog and Search won't work without it, in this case, I don't believe there is much to be done or to be fixed. It is similar to MySQL being down.

At least in this case we are allowing the customer to place the order, since the indexer shouldn't block the place order process.

What do you think?

Reply:
yes, agree we have isolated the problem, but is the stock qty and salable qty will be ok after that ?

Reply:
I guess what we could do in this case is `markIndexerAsInvalid` to make sure when ElasticSearch is back up the reindex will be done...

Reply:
Hello, @Nazar65  @gabrieldagama 
1. When you set indexers to **On Schedule** mode, stop elastic search and place an order with all available quantities.
Then all indexers still valid. On the frontend, we can't add this product to the cart anymore because of an error.
2. When we have **On Save** mode, stop elastic search and place an order with all available quantities.
Then all indexers still valid, but the product disappears from the storefront.

What do you think?